either/transforms.go

```diff
@@ -0,0 +1,54 @@
+// Package either provides cross type transformations for `mo.Either`.
+//
+// The functions provided by this package are not methods of `mo.Either` due to the lack of method type parameters
+// on methods. This is part of the design decision of the Go's generics as explained here:
+// https://go.googlesource.com/proposal/+/refs/heads/master/design/43651-type-parameters.md#No-parameterized-methods
+package either
+
+import (
+	"github.com/samber/mo"
+)
+
+// MapLeft returns a new `mo.Either` wrapping the result of applying `f` to the left value of the either.
+func MapLeft[I any, O any](f func(I) O) func(either mo.Either[I, O]) mo.Either[O, O] {
+	return func(either mo.Either[I, O]) mo.Either[O, O] {
+		if either.IsLeft() {
+			return mo.Right[O](f(either.MustLeft()))
+		}
+
+		return mo.Left[O, O](either.MustRight())
+	}
+}
+
+// MapRight returns a new `mo.Either` wrapping the result of applying `f` to the right value of the either.
+func MapRight[I any, O any](f func(O) O) func(either mo.Either[I, O]) mo.Either[I, O] {
+	return func(either mo.Either[I, O]) mo.Either[I, O] {
+		if either.IsRight() {
+			return mo.Right[I](f(either.MustRight()))
+		}
+
+		return mo.Left[I, O](either.MustLeft())
+	}
+}
```

Comment:
I have a question regarding the transformers, as far as I understand from this code the `Pipe` functions are designed for type transformations:
`goEither[int, string] -> Either[string, bool] -> Either[float64, []int]`

But the transformers are type-constrained:

MapRight: Always returns same types Either[I, O] -> Either[I, O]
MapLeft: Forces both sides to same type Either[I, O] -> Either[O, O]